### PR TITLE
Renamer attribute

### PR DESCRIPTION
**Pull request by ghost**
Renamer, a class attribute that holds a `callable(string): string` . It will be applied on all fields with empty string for the name argument.
Example:
```php
use function ucfirst;

#[Renamer("ucfirst")] class DataStructure {
}
```
- [x] Renamer attribute
- [x] Unit test
- [x] Logic implementation
- [ ] ~~Tutorial~~ (no permission to edit wiki)

### CODE REVIEW

**Comment by ghost** [comment]
100% made by my self  did not steal from other repositro

**Comment by ghost** [comment]
gthihub bad my commit unverifed The email in this signature doesn’t match the committer email. this is not true

**Comment by ghost** [comment]
```
PHPUnit 9.5.20 #StandWithUkraine

.................                                                 17 / 17 (100%)

Time: 00:00.460, Memory: 4.00 MB

OK (17 tests, 25 assertions)
```
php uyunit very good @sylvrs

**Comment by ghost** [comment]
@sylvrs pls pls make tiutorial i no able make it because of fork repository no wiki can use
pls make release tag no same as that pocketmine staff every time forget

File: tests/libMarshal/UserRenamer.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	/**
+	 * @param string $firstName
+	 * @param string $lastName
+	 * @param int $age
+	 * @param float $height
+	 * @param string[] $contacts
+	 * @param string|null $email
+	 */
+	public function __construct(
+		#[Field(name: "First-name")] public string $firstName,
```
Comment (ghost) [inline tests/libMarshal/UserRenamer.php:23]: i know this part better developer make better part
but me nooooooooooooob no can better
pls help me

**Comment by ghost** [comment]
> @sylvrs pls pls make tiutorial i no able make it because of fork repository no wiki can use pls make release tag no same as that pocketmine staff everytime forget

매번 감정적으로 나를 아프게 하는

**Comment by sylvrs** [comment]
Hmm, I think the concept it nice. However, it would be worth it to make it a more generic name. Let me ponder on some naming conventions as well as other improvements that could be added to improve the flexibility of this PR.

File: src/libMarshal/attributes/Renamer.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		$this->renamer = Closure::fromCallable($renamer);
+	}
+
+    /**
+     * @return callable(string): string
+     */
+    public function getRenamer(): callable {
+        return $this->renamer;
+    }
+}
```
Comment (sylvrs) [inline src/libMarshal/attributes/Renamer.php:35]: It seems like the formatting for this class was borked

File: tests/libMarshal/UserRenamer.php (hunk 1 above)
Comment (sylvrs) [reply tests/libMarshal/UserRenamer.php:23]: What do you need help with?

Comment (ghost) [reply tests/libMarshal/UserRenamer.php:23]: > What do you need help with?

code good tihing

**Comment by ghost** [comment]
> Hmm, I think the concept it nice. However, it would be worth it to make it a more generic name. Let me ponder on some naming conventions as well as other improvements that could be added to improve the flexibility of this PR.

ok thank you review i wait so long time

**Comment by ghost** [comment]
github finally say my commit veriifed This commit was signed with the committer’s verified signature.

**Review by Endermanbugzjfc — APPROVED** [review]

**Comment by ghost** [comment]
😭😭😭

**Pull request closed without merging**